aspnet/fundamentals/dependency-injection.rst

```diff
@@ -1,10 +1,277 @@
+
+To demonstrate the difference between these four lifetime and registration options, consider a simple interface that represents one or more tasks as an *operation* with a unique identifier, ``OperationId``. Depending on how we configure the lifetime for this service, the container will provide either the same or different instances of the service to the requesting class. To make it clear which lifetime is being requested, we will create one type per lifetime option:
+
+.. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Interfaces/IOperation.cs
+	:language: c#
+	:linenos:
+	:emphasize-lines: 7
+
+Next, in ``ConfigureServices``, each type is added to the container according to its named lifetime:
+
+.. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Startup.cs
+	:language: c#
+	:lines: 83-88
+	:linenos:
+	:dedent: 8
+
+Note that the *instance* lifetime type has been added with a known ID of ``Guid.Empty`` so it will be clear when this type is in use. We have also registered an ``OperationService`` that depends on each of the other ``Operation`` types, so that it will be clear within a request whether this service is getting the same instance as the controller, or a new one, for each operation type.
+
+.. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Services/OperationService.cs
+	:language: c#
+	:linenos:
+
+To demonstrate the object lifetimes within and between separate individual requests to the application, the sample includes an ``OperationsController`` that requests each kind of ``IOperation`` type as well as an ``OperationService``. The ``Index`` action then displays all of the controller's and service's ``OperationId`` values.
+
+.. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Controllers/OperationsController.cs
+	:language: c#
+	:linenos:
+
+Now two separate requests are made to this controller action:
+
+.. image:: dependency-injection/_static/lifetimes_request1.png
+.. image:: dependency-injection/_static/lifetimes_request2.png
+
+Observe which of the ``OperationId`` values varies within a request, and between requests.
+
+- *Transient* objects are always different; a new instance is provided to every controller and every service.
+- *Scoped* objects are the same within a request, but different across different requests
+- *Singleton* objects are the same for every object and every request
+- *Instance* objects are the same for every object and every request, and are the exact instance that was specified in ``ConfigureServices``
+
+Request Services and Application Services
```

Comment:
This section is here based on conversation with @rynowak. I've added a note, but I'm not sure that addresses your concern since it sounds like you want this to be smaller, and I'm only making it larger by adding a note. Perhaps Ryan can weigh in on the current state of this section?